syntax = "proto3";

package com.romit.workouttracker;

option java_multiple_files = true;
option java_package = "com.romit.workouttracker.proto";
option java_outer_classname = "DataAnalysisProto";

service DataAnalysisService {
  rpc AnalyzeData (PastWeekWorkouts) returns (AnalysisResult);
}
message PastWeekWorkouts {
  repeated WorkoutData workouts = 1;
}
message WorkoutData {
  int32 workout_id = 1;
  string date = 2;
  repeated WorkoutExercise exercises = 3;
}
message WorkoutExercise {
  string name = 2;
  string primary_muscle_group = 3;
  int32 sets = 4;
  int32 reps = 5;
  float weight = 6;
}
message AnalysisResult {
  string status = 1;
  Metrics results = 2;
}
message Metrics {
  int32 count = 1;
  repeated float seven_day_volume_heatmap = 2;
  map<string, float> muscle_wise_volume = 3;
}